src/Provider/FreeGeoIp/FreeGeoIp.php

```diff
@@ -83,4 +94,36 @@ public function getName(): string
     {
         return 'free_geo_ip';
     }
+
+    /**
+     * @param string $locale
+     */
+    public function setLocale(string $locale)
```

Comment:
I have also a proposal. I think that a second parameter in Common\Provider\AbstractHttpProvider is redundant and he is always overwritten in child classes. And I think this violates the Liskov Substitution Principle.